spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -641,6 +641,9 @@ private void shutdownAndWaitOrCallback(@Nullable Runnable callback) {
 			}
 			else {
 				logger.info("Shutdown ignored - container is already stopped");
+				if (callback != null) {
+					getTaskExecutor().execute(callback);
```

Comment:
There is no reason to call this callback on a different thread.

Plus, take a look, please, there is one more `return` in this method where we have to take care about this callback, too.

Reply:
You are right! I will fix this and cover the other case as soon as I get around to work on this again.